### PR TITLE
fix(gatsby): Add `directory` to GatsbyCacheLmdb

### PR DESCRIPTION
**Pull request by LekoArts**
## Description

MDX plugin uses `cache.directory` to write stuff

Same as:

https://github.com/gatsbyjs/gatsby/blob/79b7b0407ddc6b53c2e86b0cdd16866283a928d2/packages/gatsby/src/utils/cache.ts#L29

### CODE REVIEW

**Review by pieh — APPROVED** [review]
Let's ship it!

File: packages/gatsby/src/utils/cache-lmdb.ts [hunk 1]
```diff
@@ -16,9 +16,12 @@ export default class GatsbyCacheLmdb {
   private static store
   private db: Database | undefined
   public readonly name: string
+  // Needed for plugins that want to write data to the cache directory
+  public readonly directory: string
 
   constructor({ name = `db` }: { name: string }) {
     this.name = name
+    this.directory = path.join(process.cwd(), `.cache/caches/${name}`)
```
Comment (pieh) [inline packages/gatsby/src/utils/cache-lmdb.ts:24]: Oh, actually - do we need to also ensure this dir exists like in https://github.com/gatsbyjs/gatsby/blob/a0b542be35c15e71f8fe76cc5400576dfd0b1391/packages/gatsby/src/utils/cache.ts#L33 ?

Comment (LekoArts) [reply packages/gatsby/src/utils/cache-lmdb.ts:24]: Yeah, that's a good point. We probably should ensure that

Comment (LekoArts) [reply packages/gatsby/src/utils/cache-lmdb.ts:25]: @pieh added an `init` method

**Review by pieh — APPROVED** [review]
Let's ship it!

**Pull request merged**